README.md

```diff
@@ -8,131 +8,133 @@ products:
- 2. You are logged into Azure CLI. You can login using **az login** command.
- 
-Azure Active Directory Authentication will only work if the following conditions are met:
- 1. Your on-premise active directory is synced with Azure AD. 
- 2. You are running this code on a domain joined machine.   
+Windows
 
-Since your developer account has access to the Key Vault, you should see the secret on the web page. Principal Used will show type "User" and your user account. 
+```cmd
+setx KEY_VAULT_URI "<YourKeyVaultUrl>"
+```
 
-You can also use a service principal to run the application on your local development machine. See the section "Running the application using a service principal" later in the tutorial on how to do this. 
+### Deploy the Web App to Azure
+Use any of the methods outlined on [Deploy your app to Azure App Service] to publish the Web App to Azure.
 
-## Step 6: Deploy the Web App to Azure
-Use any of the methods outlined on [Deploy your app to Azure App Service](https://docs.microsoft.com/en-us/azure/app-service-web/web-sites-deploy) to publish the Web App to Azure. 
-After you deploy it, browse to the web app. You should see the secret on the web page, and this time the Principal Used will show "App", since it ran under the context of the App Service. 
-The AppId of the MSI will be displayed. 
+Step 1: Set environment variable in app service.
 
-## Summary
-The web app was successfully able to get a secret at runtime from Azure Key Vault using your developer account during development, and using MSI when deployed to Azure, without any code change between local development environment and Azure. 
-As a result, you did not have to explicitly handle a service principal credential to authenticate to Azure AD to get a token to call Key Vault. You do not have to worry about renewing the service principal credential either, since MSI takes care of that.  
+*  Search for your app service in **Search Resources dialog box**
+*  Select **Setting** > **Configuration** > **New application setting**
+*  Set the name to **KEY_VAULT_URI** and value with your **Key Vault Url**  
 
-## Running the application using a service principal in local development environment
->Note: It is recommended to use your developer context for local development, since you do not need to create or share a service principal for that. If that does not work for you, you can use a service principal, but do not check in the certificate or secret in source repos, and share them securely.
+After you deploy it, browse to the web app. You should see the secret on the web page.
 
-To run the application using a service principal in the local development environment, follow these steps
+## How to use AzureCliCredential
+There are 2 approaches to use AzureCliCredential. First way is create AzureCliCredential directly, the other way is use AzureCliCredential which is chained in DefaultAzureCredential.
```

Comment:
Nit: put code in backticks, e.g. `AzureCliCredential`.

Reply:
Resovled